### ISSUE TITLE
LELU Activation Function: Proposal for PyTorch

### ISSUE THREAD

**Issue by falseywinchnet** [pytorch/pytorch#165982]
### Summary

We propose the introduction of a modified SILU activation function to be called **LELU** — the *Logistic Error Linear Unit*. LELU serves as a lower-cost, analytically consistent alternative to GELU that is derived from the fixed-point correspondence between the logistic sigmoid and its associated cumulative distribution function (CDF) and probability density function (PDF). 

---

### Motivation

The Gaussian Error Linear Unit (GELU) is designed around the Gaussian CDF, offering smoother activation transitions than ReLU or ELU. However, GELU requires relatively costly operations (erf/tanh + cubic terms) and is only an approximation of the true Gaussian gate.

In contrast, LELU is grounded in the Logistic CDF which is analytically simpler, fully differentiable, and exhibits similar curvature to the Gaussian but with computational advantages. The logistic sigmoid family offers a natural match to many normalized training distributions observed in deep networks.

---

### Derivation Overview

LELU is defined analogously to GELU:

```
LELU(x) = x * 0.5 * (1 + tanh(a * (x)))
```

where `a = π / (2√3)` arises from the equivalence between the variance of the standard logistic distribution and that of the standard normal distribution.

---

### Proposal

Add `torch.nn.LELU` and `torch.nn.functional.lelu`

Documentation and tests can mirror the structure of `torch.nn.GELU`, with additional notes on logistic variance equivalence and performance observations.

```py

import math
import torch
import torch.nn as nn
import torch.nn.functional as F

class LELU(nn.Module):
    def __init__(self):
        super().__init__()
        val = math.pi / math.sqrt(3.0)
        val2 = 1/val
        self.register_buffer("alpha", torch.tensor(val))
        self.register_buffer("alpha_inv", torch.tensor(val2))
        self._silu = nn.SiLU(inplace=False)

    def forward(self, x: torch.Tensor) -> torch.Tensor:
            return self._silu(x * self.alpha) *self.alpha_inv
```

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by falseywinchnet** [comment]
```py
import torch
import torch.nn as nn
import math, time

# --- Define LELU ---
class LELU(nn.Module):
    def __init__(self):
        super().__init__()
        val = math.pi / math.sqrt(3.0)
        val2 = 1 / val
        self.register_buffer("alpha", torch.tensor(val))
        self.register_buffer("alpha_inv", torch.tensor(val2))
        self._silu = nn.SiLU(inplace=False)

    def forward(self, x):
        return self._silu(x * self.alpha) * self.alpha_inv

# --- Benchmarking utility ---
def benchmark(act1, act2, shape=(4096, 4096), device="cuda", repeats=50):
    x = torch.randn(shape, device=device, requires_grad=True)
    losses = []

    # Warm-up
    for _ in range(10):
        (act1(x).sum()).backward(retain_graph=True)
        x.grad.zero_()

    def time_op(act):
        torch.cuda.synchronize()
        start = time.time()
        for _ in range(repeats):
            y = act(x)
            loss = y.sum()
            loss.backward(retain_graph=True)
            x.grad.zero_()
        torch.cuda.synchronize()
        return (time.time() - start) / repeats

    t1 = time_op(act1)
    t2 = time_op(act2)
    print(f"Average time per forward+backward (shape={shape}, repeats={repeats}):")
    print(f"  {act1.__class__.__name__:<10}: {t1*1000:.3f} ms")
    print(f"  {act2.__class__.__name__:<10}: {t2*1000:.3f} ms")
    print(f"  Ratio (LELU / GELU): {t1 / t2:.3f}x")

# --- Run benchmark ---
device = "cuda" if torch.cuda.is_available() else "cpu"
benchmark(LELU().to(device), nn.GELU().to(device), device=device)

```

on T4 gpu:
Average time per forward+backward (shape=(4096, 4096), repeats=50):
  LELU      : 4.733 ms
  GELU      : 3.253 ms
  Ratio (LELU / GELU): 1.455x

on M4 mac:
Average time per forward+backward (shape=(4096, 4096), repeats=50):
  LELU      : 24.975 ms
  GELU      : 31.138 ms
  Ratio (LELU / GELU): 0.802x



owing to highly optimized kernels on NVIDIA, it is difficult to get a win- all ops are fused there.


Device=cuda, dtype=torch.float16, shape=(4096, 4096)
max |f_triton - f_ref| = 3.906e-03,  max |g_triton - g_ref| = 1.953e-03
GELU(exact)   :   1.597 ms   (x1.000)
GELU(tanh)    :   1.574 ms   (x0.985)
LELU(triton)  :   1.573 ms   (x0.985)

Device=cuda, dtype=torch.float32, shape=(4096, 4096)
max |f_triton - f_ref| = 4.768e-07,  max |g_triton - g_ref| = 7.153e-07
GELU(exact)   :   2.521 ms   (x1.000)
GELU(tanh)    :   2.525 ms   (x1.001)
LELU(triton)  :   3.102 ms   (x1.230)
